internal/util/struct.go

```diff
@@ -0,0 +1,29 @@
+package util
+
+import (
+	"errors"
+	"reflect"
+)
+
+var (
+	// ErrNoStruct indicates the passed argument is not a struct
+	ErrNoStruct = errors.New("no struct")
+)
+
+// GetFieldNames extracts all field names of struct T. Returns error if T is no struct.
+// TODO(lebogg): Only take exported fields
+func GetFieldNames[T any]() (fieldNames []string, err error) {
```

Comment:
Ideally, I would like to have sth. like `T struct` to avoid strings ect. at compile time already. Any idea? @oxisto